pkg/gds/admin.go

```diff
@@ -2092,8 +2092,17 @@ func (s *Admin) rejectRegistration(vasp *pb.VASP, reason string, claims *tokens.
 			continue
 		}
 
+		// Delete the certificate request
 		if err = s.db.DeleteCertReq(careq.Id); err != nil {
 			log.Error().Err(err).Str("id", careq.Id).Msg("could not delete certificate request")
+			continue
+		}
```

Comment:
I think we should take a minute to challenge our design and our assumptions. Is deleting the certificate request the right thing to do or should we mark it as `CR_REJECTED`? 

On the one hand, deleting the certificate request means there are fewer certificate requests for CertMan to iterate over (though we could fix that by tombstoning or archiving into a different namespace). On the other hand, if we want to "unreject" this VASP, we'll have to create a new certificate request.

To me, deleting the certificate request makes sense, but I did just want to take a minute to review our choices. 

Any thoughts one way or the other?

Reply:
In general I think I agree with deleting the certificate requests. In my mind they are more of a method of interacting with the certificate provider and once they have served their purpose there's not much use having them around besides debug purposes.

Perhaps to allow a path to "unrejection", we could mark as `CR_REJECTED` here and have the cert manager be responsible for deleting old certificate requests.